sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/MarketplacePlan.cs

```diff
@@ -47,23 +43,31 @@ public abstract partial class MachineLearningAssistConfiguration
         /// </list>
         /// </para>
         /// </summary>
-        private protected IDictionary<string, BinaryData> _serializedAdditionalRawData;
+        private IDictionary<string, BinaryData> _serializedAdditionalRawData;
 
-        /// <summary> Initializes a new instance of <see cref="MachineLearningAssistConfiguration"/>. </summary>
-        protected MachineLearningAssistConfiguration()
+        /// <summary> Initializes a new instance of <see cref="MarketplacePlan"/>. </summary>
+        internal MarketplacePlan()
         {
         }
 
-        /// <summary> Initializes a new instance of <see cref="MachineLearningAssistConfiguration"/>. </summary>
-        /// <param name="mlAssist"> [Required] Indicates whether MLAssist feature is enabled. </param>
+        /// <summary> Initializes a new instance of <see cref="MarketplacePlan"/>. </summary>
+        /// <param name="offerId"> The identifying name of the Offer of the Marketplace Plan. </param>
+        /// <param name="planId"> The identifying name of the Plan of the Marketplace Plan. </param>
+        /// <param name="publisherId"> The identifying name of the Publisher of the Marketplace Plan. </param>
         /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
-        internal MachineLearningAssistConfiguration(MLAssistConfigurationType mlAssist, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        internal MarketplacePlan(string offerId, string planId, string publisherId, IDictionary<string, BinaryData> serializedAdditionalRawData)
         {
-            MlAssist = mlAssist;
+            OfferId = offerId;
+            PlanId = planId;
+            PublisherId = publisherId;
             _serializedAdditionalRawData = serializedAdditionalRawData;
         }
 
-        /// <summary> [Required] Indicates whether MLAssist feature is enabled. </summary>
-        internal MLAssistConfigurationType MlAssist { get; set; }
+        /// <summary> The identifying name of the Offer of the Marketplace Plan. </summary>
+        public string OfferId { get; }
```

Comment:
What's the format of those ID properties?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

Reply:
Looks like it can be any string
https://msdata.visualstudio.com/Vienna/_git/vienna?path=/src/azureml-api/src/ManagementFrontEnd/Contracts/V20240701Preview/MarketplaceSubscriptions/MarketplacePlan.cs&version=GBmaster